src/daft-core/src/array/ops/concat_agg.rs

```diff
@@ -146,6 +149,43 @@ impl DaftConcatAggable for ListArray {
     }
 }
 
+impl DaftConcatAggable for DataArray<Utf8Type> {
+    type Output = DaftResult<Self>;
+
+    fn concat(&self) -> Self::Output {
+        let mut concat_result = String::new();
+
+        for idx in 0..self.len() {
+            if self.get(idx).is_some() {
+                let x = self.get(idx).unwrap().to_string();
+                concat_result.push_str(&x);
+            }
+        }
+        let result_box = Box::new(arrow2::array::Utf8Array::<i64>::from_slice([concat_result]));
+
+        DataArray::new(self.field().clone().into(), result_box)
+    }
+
+    fn grouped_concat(&self, groups: &super::GroupIndices) -> Self::Output {
+        let mut result_data = vec![];
+        for group in groups {
+            let mut group_res = String::new();
+            for idx in group {
+                let ind: usize = idx.to_usize();
+                if self.get(ind).is_some() {
+                    let x = self.get(ind).unwrap().to_string();
+                    group_res.push_str(&x);
+                }
+            }
+            result_data.push(group_res);
+        }
```

Comment:
Similar to above, let's not do `push_str`. You can see grouped_sum: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/sum.rs#L23 for inspiration on how to do this for sum via `trusted_len_iter`. But instead of summing, you would be concating the strings. You can try: https://doc.rust-lang.org/std/primitive.slice.html#method.join